From 08a51ec82618d33f369b1f39d20a1c9a297adf83 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Mon, 21 Apr 2014 09:22:24 +0100
Subject: [PATCH 196/351] mfd: arizona: Sync arizona_of_get_named_gpio with
 upstream

Sync up arizona_of_get_named_gpio with the upstream implementation. Note
that this patch updates all the callers of this function as well.

Change-Id: I6654a1eb0aa70d0941f11cdfd4c9960a642961ee
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/extcon/extcon-arizona.c  |    5 +++--
 drivers/mfd/arizona-core.c       |   30 +++++++++++++-----------------
 drivers/regulator/arizona-ldo1.c |    2 +-
 drivers/switch/switch-arizona.c  |    5 +++--
 include/linux/mfd/arizona/core.h |    2 +-
 5 files changed, 21 insertions(+), 23 deletions(-)

diff --git a/drivers/extcon/extcon-arizona.c b/drivers/extcon/extcon-arizona.c
index 5238ec1..49a1065 100644
--- a/drivers/extcon/extcon-arizona.c
+++ b/drivers/extcon/extcon-arizona.c
@@ -1192,8 +1192,9 @@ static int arizona_extcon_get_pdata(struct arizona *arizona)
 	arizona_of_read_u32(arizona, "wlf,micd-detect-debounce", false,
 			    &pdata->micd_detect_debounce);
 
-	arizona_of_get_named_gpio(arizona, "wlf,micd-pol-gpio", false,
-				  &pdata->micd_pol_gpio);
+	pdata->micd_pol_gpio = arizona_of_get_named_gpio(arizona,
+							 "wlf,micd-pol-gpio",
+							 false);
 
 	arizona_of_read_u32(arizona, "wlf,micd-bias-start-time", false,
 			    &pdata->micd_bias_start_time);
diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index 89a3bc1..33a89e5 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -620,26 +620,22 @@ int arizona_of_get_type(struct device *dev)
 }
 EXPORT_SYMBOL_GPL(arizona_of_get_type);
 
-int arizona_of_get_named_gpio(struct arizona *arizona,
-			      const char *prop, bool mandatory,
-			      int *gpio)
+int arizona_of_get_named_gpio(struct arizona *arizona, const char *prop,
+			      bool mandatory)
 {
-	int ret;
-
-	ret = of_get_named_gpio(arizona->dev->of_node, prop, 0);
-	*gpio = ret;
-
-	if (ret >= 0)
-		return ret;
+	int gpio;
 
-	*gpio = 0;
+	gpio = of_get_named_gpio(arizona->dev->of_node, prop, 0);
 
-	if (mandatory)
-		dev_err(arizona->dev,
-			"Mandatory DT gpio %s missing/malformed: %d\n",
-			prop, ret);
+	if (gpio < 0) {
+		if (mandatory)
+			dev_err(arizona->dev,
+				"Mandatory DT gpio %s missing/malformed: %d\n",
+				prop, gpio);
+		gpio = 0;
+	}
 
-	return ret;
+	return gpio;
 }
 EXPORT_SYMBOL_GPL(arizona_of_get_named_gpio);
 
@@ -846,7 +842,7 @@ static int arizona_of_get_core_pdata(struct arizona *arizona)
 {
 	struct arizona_pdata *pdata = &arizona->pdata;
 
-	arizona_of_get_named_gpio(arizona, "wlf,reset", true, &pdata->reset);
+	pdata->reset = arizona_of_get_named_gpio(arizona, "wlf,reset", true);
 
 	arizona_of_get_micd_ranges(arizona, "wlf,micd-ranges");
 	arizona_of_get_micd_configs(arizona, "wlf,micd-configs");
diff --git a/drivers/regulator/arizona-ldo1.c b/drivers/regulator/arizona-ldo1.c
index daccca6..57bff05 100644
--- a/drivers/regulator/arizona-ldo1.c
+++ b/drivers/regulator/arizona-ldo1.c
@@ -185,7 +185,7 @@ static int arizona_ldo1_of_get_pdata(struct arizona *arizona,
 	struct device_node *init_node, *dcvdd_node;
 	struct regulator_init_data *init_data;
 
-	arizona_of_get_named_gpio(arizona, "wlf,ldoena", true, &pdata->ldoena);
+	pdata->ldoena = arizona_of_get_named_gpio(arizona, "wlf,ldoena", true);
 
 	init_node = of_get_child_by_name(arizona->dev->of_node, "ldo1");
 	dcvdd_node = of_parse_phandle(arizona->dev->of_node, "DCVDD-supply", 0);
diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index e610765..58c3e3e 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -1348,8 +1348,9 @@ static int arizona_extcon_get_pdata(struct arizona *arizona)
 	arizona_of_read_u32(arizona, "wlf,micd-detect-debounce", false,
 			    &pdata->micd_detect_debounce);
 
-	arizona_of_get_named_gpio(arizona, "wlf,micd-pol-gpio", false,
-				  &pdata->micd_pol_gpio);
+	pdata->micd_pol_gpio = arizona_of_get_named_gpio(arizona,
+							 "wlf,micd-pol-gpio",
+							 false);
 
 	arizona_of_read_u32(arizona, "wlf,micd-bias-start-time", false,
 			    &pdata->micd_bias_start_time);
diff --git a/include/linux/mfd/arizona/core.h b/include/linux/mfd/arizona/core.h
index 3e2499f..a2c9f06 100644
--- a/include/linux/mfd/arizona/core.h
+++ b/include/linux/mfd/arizona/core.h
@@ -130,7 +130,7 @@ int florida_patch(struct arizona *arizona);
 int wm8997_patch(struct arizona *arizona);
 
 extern int arizona_of_get_named_gpio(struct arizona *arizona, const char *prop,
-				     bool mandatory, int *gpio);
+				     bool mandatory);
 extern int arizona_of_read_u32_array(struct arizona *arizona, const char *prop,
 				     bool mandatory, u32 *data, size_t num);
 extern int arizona_of_read_u32(struct arizona *arizona, const char* prop,
-- 
1.7.9.5

